### ISSUE TITLE
Getting input from user with 'terraform plan' fails

### ISSUE THREAD

**Issue by pete0emerson** [hashicorp/terraform#3083]
# Terraform version

```
$ terraform version
Terraform v0.6.3
```

# Description

If a variable isn't set, then according to https://www.terraform.io/intro/getting-started/variables.html:

> If you execute terraform plan or apply without doing anything, Terraform will ask you to input the variables interactively.

Using `terraform apply` works:

```
$ terraform apply
var.aws_access_key
  AWS access key

  Enter a value:

```

But `terraform plan` does not:

```
 $ terraform plan
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * 4 error(s) occurred:

* Required variable not set: aws_secret_key
* Required variable not set: aws_region
* Required variable not set: cidr_block
* Required variable not set: aws_access_key
```

# Steps to reproduce

In `vpc.tf`:

```
provider "aws" {
    access_key = "${var.aws_access_key}"
    secret_key = "${var.aws_secret_key}"
    region = "${var.aws_region}"
}

resource "aws_vpc" "dev" {
    cidr_block = "${var.cidr_block}"
    instance_tenancy = "default"
    tags {
        Name = "dev"
    }
}
```

In `variables.tf`:

```
variable "aws_access_key" {
    description = "AWS access key"
}

variable "aws_secret_key" {
    description = "AWS secret key"
}
variable "aws_region" {
    description = "AWS region"
}
variable "cidr_block" {
    description = "CIDR block"
}
```

**Comment by RoryHunter** [comment]
Just ran into the same issue. Version v0.6.3.

**Comment by jagadeeshaluri00** [comment]
is there any work around for this until it's officially fixed?

**Comment by jagadeeshaluri00** [comment]
looks like TF_VAR_access_key and TF_VAR_secret_key has been removed in latest release. that seem to be the reason for my issues. once I started using AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY , it's working as expected.

**Comment by jen20** [comment]
Hi @pete0emerson! It looks like this is a duplicate of #3737, which was subsequently fixed in #3843. That fix is currently on the `master` branch and will be in the next point release, so I'll go ahead and close this issue for now - if I'm misunderstanding the report then please feel free to re-open it.
